### PR TITLE
add label smooth loss for NPU

### PR DESCRIPTION
**Pull request by HydrogenSulfate**
- 背景：
由于NPU上的CrossEntropyLoss暂时缺少对soft_label的情况处理，而label smooth需要使用soft_label这一功能，因此将label smooth拆分为两个loss：均匀分布的标签手动计算交叉熵+硬标签调用CrossEntropyLoss计算交叉熵，然后按公式 加权求和。
- 验证：
  ```python
  import paddle
  import paddle.nn.functional as F
  
  ls_eps = 0.618
  num_classes = 400
  B = 16
  
  
  def label_smooth_loss(scores, labels, **kwargs):
      """[summary]
  
      Args:
      scores ([type]): [N,K]
      labels ([type]): [N,]
  
      Returns:
      [type]: [description]
      """
      labels = F.one_hot(labels, num_classes)
      labels = F.label_smooth(labels, epsilon=ls_eps)
      labels = paddle.squeeze(labels, axis=1)
      loss = F.cross_entropy(scores, labels, soft_label=True, **kwargs)
      return loss
  
  def label_smooth_loss_npu(scores, labels, **kwargs):
      """不需要使用soft_label参数的label smooth loss，将计算方式拆成硬标签loss和均匀分布loss加权求和的形式
  
      Args:
      scores (paddle.Tensor): [N, num_classes]
      labels (paddle.Tensor): [N, ]
  
      Returns:
      paddle.Tensor: [1,]
      """
      num_classes = scores.shape[-1]
      hard_loss = (1.0 - ls_eps) * F.cross_entropy(scores, labels)
      uniform_loss = (ls_eps / num_classes) * (-F.log_softmax(scores, -1).sum(-1).mean(0))
      loss = hard_loss + uniform_loss
      return loss
  
  scores = paddle.randn([B, num_classes])
  labels = paddle.randint(0, num_classes, [B, ])
  
  sm_loss1 = label_smooth_loss(scores, labels)
  sm_loss2 = label_smooth_loss_npu(scores, labels)
  
  print(sm_loss1)
  print(sm_loss2)
  print(paddle.allclose(sm_loss1, sm_loss2))
  ```
  样例输出：
  ```bash
  Tensor(shape=[1], dtype=float32, place=CUDAPlace(0), stop_gradient=True,
         [6.33457518])
  Tensor(shape=[1], dtype=float32, place=CUDAPlace(0), stop_gradient=True,
         [6.33457470])
  Tensor(shape=[1], dtype=bool, place=CUDAPlace(0), stop_gradient=True,
         [True])
  ```

### CODE REVIEW

**Comment by rgn9536903** [comment]
Score 20000 help me please

**Comment by huangjun12** [comment]
LGTM
